paperqa/agents/env.py

```diff
@@ -39,17 +40,17 @@ def settings_to_tools(
     embedding_model: EmbeddingModel | None = POPULATE_FROM_SETTINGS,
 ) -> list[Tool]:
     """
-    Convert a Settings into tools, confirming the gen_answer tool is present.
+    Convert a Settings into tools, confirming the complete tool is present.
 
-    NOTE: the last element of the return will always be GenerateAnswer.
+    NOTE: the last element of the return will always be Complete.
     """
     llm_model = llm_model or settings.get_llm()
     summary_llm_model = summary_llm_model or settings.get_summary_llm()
     embedding_model = embedding_model or settings.get_embedding_model()
     tools: list[Tool] = []
-    has_answer_tool = False
+    has_complete_tool = False
```

Comment:
Is this check really necessary? If any of the tools are missing, it will fail. No PaperSearch means no papers, no GatherEvidence means no contexts, etc.

Reply:
Ah I see - it's for the special zero-shot setting.

Reply:
Sure, I made https://github.com/Future-House/paper-qa/pull/685 to keep this PR minimal

Reply:
> Ah I see - it's for the special zero-shot setting.

The validation is really to protect:

- Clients who are doing edge case things (e.g. using a subset of the tools, and they should also override the `agent_prompt`)
- Unit tests from failing for unexpected reasons
    - However, this validation usually complicates life for unit test writers, not makes it better

Reply:
Alright, `git rebase`d atop https://github.com/Future-House/paper-qa/pull/685, resolving this one